### PR TITLE
Implementation of BlandAltman comparer with new API

### PR DESCRIPTION
**Pull request by wptmdoorn**
This pull requests implements the BlandAltman class using the new Comparer API. It also passes all units test.

Happy to receive feedback!

### CODE REVIEW

File: methcomp/blandaltman.py [hunk 1]
```diff
@@ -181,74 +182,77 @@ def plot(
...
 
         if reference:
             ax.axhline(0, color="grey", linestyle="-", alpha=0.4)
 
         # confidence intervals (if requested)
         if self.CI is not None:
-            ax.axhspan(self.CI_mean[0], self.CI_mean[1], color=color_mean, alpha=0.2)
-            ax.axhspan(self.CI_upper[0], self.CI_upper[1], color=color_loa, alpha=0.2)
-            ax.axhspan(self.CI_lower[0], self.CI_lower[1], color=color_loa, alpha=0.2)
+            ax.axhspan(*mean_CI, color=color_mean, alpha=0.2)
```
Comment (thomasburgess) [inline methcomp/blandaltman.py:203]: should let this alpha be `self.CI_alpha`, initialized by default to `0.2` in constructor

File: methcomp/blandaltman.py [hunk 2]
```diff
@@ -181,74 +182,77 @@ def plot(
...
+        sd_diff = self.result["sd_diff"]
 
         # individual points
         ax.scatter(self.mean, self.diff, **pkws)
 
         # mean difference and SD lines
         ax.axhline(mean, color=color_mean, linestyle="-")
-        ax.axhline(mean + self.loa_sd, color=color_loa, linestyle="--")
-        ax.axhline(mean - self.loa_sd, color=color_loa, linestyle="--")
+        ax.axhline(loa_upper, color=color_loa, linestyle="--")
```
Comment (thomasburgess) [inline methcomp/blandaltman.py:195]: should let this alpha be `self.loa_linestyle`, initialized by default `"--"` in constructor

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -18,7 +18,7 @@ repos:
     rev: 21.5b1
     hooks:
     -   id: black
-        language_version: python3.8
+        language_version: python
```
Comment (thomasburgess) [inline .pre-commit-config.yaml:21]: huh, I didn't even know that was an option :), if it  works it works...

File: methcomp/blandaltman.py [hunk 4]
```diff
@@ -48,16 +50,29 @@ def __init__(
...
+        if self.CI is not None and (self.CI > 1 or self.CI < 0):
+            raise ValueError("Confidence interval must be between 0 and 1.")
 
-    def compute(self) -> Dict[str, Any]:
+        if self.diff_method not in ["absolute", "percentage"]:
+            raise ValueError(
+                "The provided difference method must be either absolute or percentage."
+            )
+
+        # if any([not isinstance(x, str) for x in [self.x_title, self.y_title]]):
```
Comment (thomasburgess) [inline methcomp/blandaltman.py:72]: Should probably delete commented out block

**Review by thomasburgess — COMMENTED** [review]
Looking pretty good to me 👍

File: methcomp/blandaltman.py (hunk 4 above)
Comment (wptmdoorn) [reply methcomp/blandaltman.py:72]: resolved

File: .pre-commit-config.yaml (hunk 3 above)
Comment (wptmdoorn) [reply .pre-commit-config.yaml:21]: had some problems if you have another python version, this works fine for me (so I think its OK)

Comment (thomasburgess) [reply .pre-commit-config.yaml:21]: I think it works, let's leave it

File: methcomp/blandaltman.py (hunk 1 above)
Comment (wptmdoorn) [reply methcomp/blandaltman.py]: added this now as a plot() argument since it's only graphically relevant

File: methcomp/blandaltman.py (hunk 2 above)
Comment (wptmdoorn) [reply methcomp/blandaltman.py]: added this now as a plot() argument since it's only graphically relevant

**Comment by wptmdoorn** [comment]
I've revised your comments. You can accept it and merge if you are satisfied. Thanks a lot :)

File: methcomp/blandaltman.py [hunk 5]
```diff
@@ -138,6 +131,8 @@ def plot(
         color_loa: str = "#FF7000",
         color_points: str = "#000000",
         point_kws: Dict = None,
+        ci_alpha: float = 0.2,
```
Comment (thomasburgess) [inline methcomp/blandaltman.py:134]: This solution is better than what I suggested.

**Review by thomasburgess — APPROVED** [review]
Looking good from my side

**Review by thomasburgess — APPROVED** [review]

**Pull request merged**